### PR TITLE
Replace empty string literals in Marshal.cs

### LINKED ISSUE #68347: Empty string literals in Marshal.cs
While working on #34056 I noticed that for some exceptions inside `System.Runtime.InteropServices.Marshal` some exceptions get thrown with empty string literals. Wouldn't it be better for these empty string literals to be replaced with the string.Empty constant?

e.g.
image

### PR DESCRIPTION
**Pull request by tudoranghelina91**
Replace emtpy string literals exception messages with null in Marshal.cs

Fix #68347

### CODE REVIEW

**Comment by tudoranghelina91** [comment]
@dotnet-policy-service agree

File: src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.cs [hunk 1]
```diff
@@ -798,7 +798,7 @@ public static IntPtr GetHINSTANCE(Module m)
                     return new System.NotImplementedException();
                 //case HResults.E_POINTER:
                 case HResults.RO_E_CLOSED:
```
Comment (jkotas) [inline src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.cs:800]:
```suggestion
                case HResults.COR_E_OBJECTDISPOSED:
                case HResults.RO_E_CLOSED:
```
We can share the implementation for both cases.

File: src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.cs [hunk 2]
```diff
@@ -739,7 +739,7 @@ public static IntPtr GetHINSTANCE(Module m)
                 case HResults.E_POINTER:
                     return new System.NullReferenceException();
                 case HResults.COR_E_OBJECTDISPOSED:
```
Comment (jkotas) [inline src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.cs:741]:
```suggestion
```

File: src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.cs [hunk 3]
```diff
@@ -739,7 +739,7 @@ public static IntPtr GetHINSTANCE(Module m)
...
-                    return new System.ObjectDisposedException("");
+                    return new System.ObjectDisposedException(null);
```
Comment (jkotas) [inline src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.cs:742]:
```suggestion
```

File: src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.cs [hunk 4]
```diff
@@ -824,7 +824,7 @@ public static IntPtr GetHINSTANCE(Module m)
                 case HResults.TYPE_E_TYPEMISMATCH:
                 case HResults.CO_E_NOTINITIALIZED:
                 case HResults.RPC_E_CHANGED_MODE:
```
Comment (jkotas) [inline src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.cs:826]: Same here. The whole block of cases can be deleted. The `default:` case is going to take care of this just fine. (github does not allow me to make this as a suggestion.)

File: src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.cs [hunk 5]
```diff
@@ -798,33 +796,12 @@ public static IntPtr GetHINSTANCE(Module m)
...
                 //case HResults.E_POINTER:
                 case HResults.RO_E_CLOSED:
-                    return new System.ObjectDisposedException("");
-                case HResults.COR_E_ABANDONEDMUTEX:
-                case HResults.COR_E_AMBIGUOUSIMPLEMENTATION:
-                case HResults.COR_E_CANNOTUNLOADAPPDOMAIN:
-                case HResults.COR_E_CONTEXTMARSHAL:
+                case HResults.COR_E_OBJECTDISPOSED:
+                    return new System.ObjectDisposedException(null);
                 //case HResults.COR_E_HOSTPROTECTION:
```
Comment (jkotas) [inline src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.cs:801]:
```suggestion
```

File: src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.cs [hunk 6]
```diff
@@ -798,33 +796,12 @@ public static IntPtr GetHINSTANCE(Module m)
...
-                case HResults.COR_E_INSUFFICIENTMEMORY:
-                case HResults.COR_E_INVALIDCOMOBJECT:
-                case HResults.COR_E_KEYNOTFOUND:
-                case HResults.COR_E_MISSINGSATELLITEASSEMBLY:
-                case HResults.COR_E_SAFEARRAYRANKMISMATCH:
-                case HResults.COR_E_SAFEARRAYTYPEMISMATCH:
                 //case HResults.COR_E_SAFEHANDLEMISSINGATTRIBUTE:
                 //case HResults.COR_E_SEMAPHOREFULL:
                 //case HResults.COR_E_THREADSTOP:
```
Comment (jkotas) [inline src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.cs:804]:
```suggestion
```

**Review by jkotas — APPROVED** [review]
Thanks!

**Comment by danmoseley** [comment]
@jkotas  would we be open to API proposal to expose the parameterless ctor - that would tidy this up?

**Comment by jkotas** [comment]
We want to encourage callers to pass in the extra information about the object that triggered the exception. The situations where this extra information is not available are extremely rare. I do not see a problem with callers having to pass in `null` explicitly in these rare cases.

**Comment by jkotas** [comment]
There are no observable public API behavior changes. The ObjectName property getter [replaces `null` with empty string](https://github.com/dotnet/runtime/blob/a54a823ece1094dd05b7380614bd43566834a8f9/src/libraries/System.Private.CoreLib/src/System/ObjectDisposedException.cs#L99) and so the ObjectName property is going to return the empty string both before and after your change.

If there were observable public API behavior changes, it is unlikely we would accept the PR.

> Also, I see there are some failing checks in the pipeline

The failures are known issues. "Build Analysis" has links to github issues that are tracking these failures.

**Pull request merged**